wordpress-mobile/WordPress-iOS
# [Stats Refresh] Add Insights Tags and Categories card

Fixes #10744 

This primarily adds the Insights Tags and Categories card. 

Secondarily, it also includes:
- Some (hopefully temporary) logic to swizzle data Strings so the abbreviated numbers can be displayed. Namely:
  - Tags and Categories
  - Follower Totals
  - Comments
  - Followers > total followers.
  - Publicize
- Some refactoring to DRY out the `UITableViewCell` code.

Disclaimer:
- Posts with multiple tags/categories are displayed as a single row, and do not open a webView when selected. A future task (https://github.com/wordpress-mobile/WordPress-iOS/issues/10832) will add functionality to make those drop-downs. Example:
<img width="350" alt="grouped" src="https://user-images.githubusercontent.com/1816888/51218158-622b6080-18e8-11e9-8133-26a0ba7c2c9f.png">

To test:

---
Tags and Categories:
- On a site with T & C, go to Stats > Insights. (Hogwarts and en.blog are good ones.)
- Verify:
  - Subtitles are shown.
  - Max of 6 rows are displayed.
  - Rows have a disclosure and data bar.
  - If there are more than 6 data rows, the cell has a View more row.
  - Tapping a row opens a webView for that tag/category.
  - Tapping View more will display a temporary alert.

<img width="350" alt="t_and_c" src="https://user-images.githubusercontent.com/1816888/51218391-6b68fd00-18e9-11e9-8e20-ae739d1898b9.png">

---
No Tags and Categories:
- On a site with _no_ T & C, go to Stats > Insights.
- Verify:
  - Subtitles are _not_ shown.
  - The cell has only a `No data yet` row.

<img width="350" alt="no_t_and_c" src="https://user-images.githubusercontent.com/1816888/51218790-32318c80-18eb-11e9-8f7e-09ef1ff98e7f.png">

---
Now with abbreviated totals!
- Just a sanity check - these cards should now display abbreviated totals:
  - Follower Totals
  - Comments
  - Followers > total followers.
  - Publicize

Example:
<img width="350" alt="follower_totals" src="https://user-images.githubusercontent.com/1816888/51218938-c56ac200-18eb-11e9-82e4-f4a3c8a411b1.png">
